handlers/authentication.go

```diff
@@ -217,6 +254,79 @@ func performOpenIdAuthentication(w http.ResponseWriter, r *http.Request) bool {
 	return true
 }
 
+func performHeaderAuthentication(w http.ResponseWriter, r *http.Request) bool {
+	authInfo := getTokenStringFromHeader(r)
+
+	if authInfo == nil {
+		RespondWithError(w, http.StatusInternalServerError, "Token is empty.")
+		return false
+	}
+
+	kialiToken, err := kubernetes.GetKialiToken()
+
+	if err != nil {
+		RespondWithError(w, http.StatusInternalServerError, err.Error())
+		return false
+	}
+
+	business, err := business.Get(&api.AuthInfo{Token: kialiToken})
+	if err != nil {
+		RespondWithDetailedError(w, http.StatusInternalServerError, "Error instantiating the business layer", err.Error())
+		return false
+	}
+
+	// Get the subject for the token to validate it as a valid token
+	subjectFromToken, err := business.TokenReview.GetTokenSubject(authInfo)
+
+	if err != nil {
+		RespondWithError(w, http.StatusInternalServerError, err.Error())
+		return false
+	}
+
+	// Now that we know that the ServiceAccount token is valid, parse/decode it to extract
```

Comment:
Is this comment correct? Is it a "ServiceAccount" token or a token from the header? And where was it checked for validity (I guess my question is what does it mean to be valid? Does "valid" mean it can be used to successfully make requests with? Or "valid" as in "it is in there and it isn't nil"?

Reply:
this comment should (and has) been updated.  I started by duplicating the `performTokenAuthentication` function and this comment was a left-over of that.  Updated the comment.